lib/client/conntest/ssh.go

```diff
@@ -49,30 +68,218 @@ func NewSSHConnectionTester(cd services.ConnectionsDiagnostic) *SSHConnectionTes
 //   - the SSH Node will append a trace indicating if the has access (RBAC)
 //   - the SSH Node will append a trace indicating if the requested principal is valid for the target Node
 func (s *SSHConnectionTester) TestConnection(ctx context.Context, req TestConnectionRequest) (types.ConnectionDiagnostic, error) {
-	id := uuid.NewString()
-	connectionDiagnostic, err := types.NewConnectionDiagnosticV1(id, map[string]string{},
+	if req.ResourceKind != types.KindNode {
+		return nil, trace.BadParameter("invalid value for ResourceKind, expected %q got %q", types.KindNode, req.ResourceKind)
+	}
+
+	connectionDiagnosticID := uuid.NewString()
+	connectionDiagnostic, err := types.NewConnectionDiagnosticV1(connectionDiagnosticID, map[string]string{},
 		types.ConnectionDiagnosticSpecV1{
-			Message: types.DiagnosticMessageWaiting,
+			// We start with a failed state so that we don't need an extra update when returning non-happy paths.
+			// For the happy path, we do update the Message to Success.
+			Message: types.DiagnosticMessageFailed,
 		})
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	if err := s.cd.CreateConnectionDiagnostic(ctx, connectionDiagnostic); err != nil {
+	if err := s.clt.CreateConnectionDiagnostic(ctx, connectionDiagnostic); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	key, err := client.GenerateRSAKey()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	currentUser, err := s.clt.GetCurrentUser(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	certs, err := s.clt.GenerateUserCerts(ctx, proto.UserCertsRequest{
+		PublicKey:              key.MarshalSSHPublicKey(),
+		Username:               currentUser.GetName(),
+		Expires:                time.Now().Add(time.Minute).UTC(),
+		ConnectionDiagnosticID: connectionDiagnosticID,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
```

Comment:
It is worth to consider if we should support  `TestConnection` for trusted cluster and with MFA flow

Reply:
I would say no at this point.
We are targeting the first resource the user is adding to its cluster, to reach the "aha moment" faster.
At this point, they shouldn't have trusted clusters.
Not sure about the MFA flow :thinking: but I guess that's a no as well.

@r0mant Any opinion on this?